class.jetpack.php

```diff
@@ -3891,7 +3903,30 @@ function maybe_redirect_back_to_calypso() {
 					&& $parsed_jetpack_connect_site->path == $parsed_site->path
 					&& ( time() - $this->JETPACK_CONNECT_TIMEOUT ) < strtotime( $jetpack_connect_request['date'] ) ) {
 					// the user started a flow from calypso registering this site url in the last 24 hours
-					return true;
+
+					$redirect_data = array(
+						'direct' => false,
+						'url' => $this->JETPACK_CONNECT_FLOW_URL,
+					);
+					$endpoint = $this->JETPACK_CONNECT_REDIRECT_INFO . $user->user_id;
+					$response = Jetpack_Client::wpcom_json_api_request_as_blog( $endpoint, '1.1', array( 'method' => 'GET' ) );
+					// wp_remote_get( $this->JETPACK_CONNECT_REDIRECT_INFO . $user->user_id, array( 'timeout' => 2 ) );
+					if ( is_array( $response ) ) {
+						$redirect_data[ 'direct' ] = $response['body']['direct'];
+						$redirect_data[ 'url' ] = $response['body']['url'];
+
+					}
+					echo $response .'\n'.json_encode( $response);
```

Comment:
oh, no, of course! sorry, I should have changed the status of this to "in progress", because I'm still figuring out how to do a signed request to .com API, and that echo is part of my investigations :)